s3-commons/src/main/java/io/aiven/kafka/connect/s3/S3Utility.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.s3;
+
+import java.util.Objects;
+
+import io.aiven.kafka.connect.iam.AwsCredentialProviderFactory;
+
+import com.amazonaws.PredefinedClientConfigurations;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.retry.PredefinedBackoffStrategies;
+import com.amazonaws.retry.PredefinedRetryPolicies;
+import com.amazonaws.retry.RetryPolicy;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+
+public class S3Utility {
+
+    private final AwsCredentialProviderFactory credentialFactory = new AwsCredentialProviderFactory();
+
+    public AmazonS3 createAmazonS3Client(final S3SinkBaseConfig config) {
```

Comment:
If we move those above configs to common config, then this s3 utility can be neutral to source and sink